packages/proxy/edge/index.ts

```diff
@@ -64,9 +72,29 @@ async function handleOptions(request: Request) {
 
 export function EdgeProxyV1(opts: ProxyOpts) {
   const meterProvider = opts.meterProvider;
+  const whitelist = opts.whitelist || defaultWhitelist;
   return async (request: Request, ctx: EdgeContext) => {
-    if (request.method === "OPTIONS") {
-      return handleOptions(request);
+    // If the host is not in the whitelist, return a 403.
+    const origin = request.headers.get("origin");
+    if (
+      opts.cors &&
+      (!origin ||
+        !whitelist.some(
+          (w) =>
+            w === origin ||
+            (typeof w !== "string" && w instanceof RegExp && w.test(origin)),
```

Comment:
Why do we need `typeof w !== string` if we have `w instanceof RegExp`?

Reply:
We do not